pkg/operator/ceph/object/spec.go

```diff
@@ -111,22 +111,34 @@ func (c *clusterConfig) makeRGWPodSpec(rgwConfig *rgwConfig) (v1.PodTemplateSpec
 
 	// Replace default unreachable node toleration
 	k8sutil.AddUnreachableNodeToleration(&podSpec)
-
 	// Set the ssl cert if specified
-	if c.store.Spec.Gateway.SSLCertificateRef != "" {
+	if c.store.Spec.Gateway.SecurePort != 0 {
 		// Keep the SSL secret as secure as possible in the container. Give only user read perms.
 		// Because the Secret mount is owned by "root" and fsGroup breaks on OCP since we cannot predict it
 		// Also, we don't want to change the SCC for fsGroup to RunAsAny since it has a major broader impact
 		// Let's open the permissions a bit more so that everyone can read the cert.
 		userReadOnly := int32(0444)
+		var secretName string
+		var itemList []v1.KeyToPath
+		if c.store.Spec.Gateway.SSLCertificateRef != "" {
+			secretName = c.store.Spec.Gateway.SSLCertificateRef
+			itemList = []v1.KeyToPath{{Key: certKeyName, Path: certFilename, Mode: &userReadOnly}}
+		} else if len(c.store.Spec.GetServiceServingCert()) != 0 {
+			secretName = c.store.Spec.GetServiceServingCert()
+			itemList = []v1.KeyToPath{
+				{Key: "tls.crt", Path: certFilename, Mode: &userReadOnly},
+				{Key: "tls.key", Path: certKeyFileName, Mode: &userReadOnly},
+			}
+		} else {
+			return v1.PodTemplateSpec{}, errors.New("No ssl certs found")
+		}
```

Comment:
What if all of this were a helper function like `generateTlsCertSecret()` and some unit tests for it?